manifests/record/ptr/by_ip.pp

```diff
@@ -0,0 +1,86 @@
+# == Definition: dns::record::ptr::by_ip
+#
+# This type is defined to allow the dns::record::a resource type to
+# use pre-4.x 'iteration' to create reverse ptr records for hosts with
+# multiple ip addresses.
+#
+# === Parameters
+#
+# * `$ip`
+# The ip address for the ptr record.  Defaults to the resource title.
+#
+# * `$host`
+# The (unqualified) host pointed to by this ptr record.
+#
+# * `$ttl`
+# The TTL of the records to be created.  Defaults to undefined.
+#
+# * `$zone`
+# The domain of the host pointed to by this ptr record.
+#
+# === Actions
+#
+# Reformats the $ip, $host, $zone parameters to create a `dns::record::ptr` 
+# resource
+#
+# === Examples
+#
+# @example
```

Comment:
I have no idea - it's part of the standard puppet module template that `puppet module generate` produces, so I've gotten in the habit of including it.